src/mlpack/methods/ann/rbm/rbm.hpp

```diff
@@ -0,0 +1,447 @@
+/**
+ * @file rbm.hpp
+ * @author Kris Singh
+ * @author Shikhar Jaiswal
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license. You should have received a copy of the
+ * 3-clause BSD license along with mlpack. If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_ANN_RBM_RBM_HPP
+#define MLPACK_METHODS_ANN_RBM_RBM_HPP
+
+#include <mlpack/core.hpp>
+#include <mlpack/methods/ann/rbm/rbm_policies.hpp>
+
+namespace mlpack {
+namespace ann /** Artificial Neural Network. */ {
+
+/**
+ * The RBM class.
+ *
+ * @tparam InitializationRuleType Rule used to initialize the network.
+ * @tparam DataType The type of matrix to be used.
+ * @tparam PolicyType The RBM variant to be used (BinaryRBM or SpikeSlabRBM).
+ */
+template<
+  typename InitializationRuleType,
+  typename DataType = arma::mat,
+  typename PolicyType = BinaryRBM
+>
+class RBM
+{
+ public:
+  using NetworkType = RBM<InitializationRuleType, DataType, PolicyType>;
+  typedef typename DataType::elem_type ElemType;
+
+  /**
+   * Initialize all the parameters of the network using initializeRule.
+   *
+   * @tparam initializeRule InitializationRule object for
```

Comment:
This isn't a template parameter, also ``predictors`` should be swapped with ``initializeRule``.